### PR TITLE
deterministic deployment added via create3

### PR DESCRIPTION
**Pull request by shivam-ef**

### CODE REVIEW

File: src/user-safe/UserSafeFactory.sol [hunk 1]
```diff
@@ -17,13 +18,27 @@ contract UserSafeFactory is UpgradeableBeacon {
...
-        address safe = address(new BeaconProxy(address(this), data));
+    function createUserSafe(bytes memory saltData, bytes memory data) external returns (address) {
+        // address safe = address(new BeaconProxy(address(this), data));
+        address safe = address(
+            CREATE3.deployDeterministic(
+                abi.encodePacked(
+                    type(BeaconProxy).creationCode, 
+                    abi.encode(address(this), data)
+                ), 
+                keccak256(abi.encode(saltData, data))
```
Comment (seongyun-ko) [inline src/user-safe/UserSafeFactory.sol:38]: why do we need both `saldData` and `data`? not just `data`? 

https://github.com/Vectorized/solady/blob/main/src/utils/CREATE3.sol#L59

Comment (shivam-ef) [reply src/user-safe/UserSafeFactory.sol:38]: Salt data will basically contain user's identification like UUID from the backend. Data is the init data for the contract that is being deployed. Do you want me to use do something like abi.encode(saltData, data) and send it as one param? Can do that but will increase some gas!

Comment (seongyun-ko) [reply src/user-safe/UserSafeFactory.sol:38]: all good :)

**Review by seongyun-ko — APPROVED** [review]
lgtm

**Pull request merged**